### PR TITLE
added match_raises a generalized version of check_raises

### PR DESCRIPTION
**Pull request by JoanThibault**
proposal for issue 88: https://github.com/mirage/alcotest/issues/88

### CODE REVIEW

**Review by dinosaure — APPROVED** [review]
The CI error seems unrelated to this PR.

**Comment by samoht** [comment]
Could you run `ocamlformat` on your patch? ie. `dune build @fmt --auto` should fix the CI error.

**Comment by MisterDA** [comment]
CI failure is unrelated, and fixed by #387. Thanks for the PR!

File: src/alcotest-engine/test.ml [hunk 1]
```diff
@@ -239,5 +239,18 @@ let check_raises ?here ?pos msg exn f =
...
 
+let match_raises ?here ?pos msg exnp f =
+  show_assert msg;
+  match collect_exception f with
+  | None ->
+      check_err (fun ppf () ->
+          Fmt.pf ppf "%t%a %s: got nothing." (pp_location ?here ?pos) Pp.tag
+            `Fail msg)
+  | Some e ->
+      if exnp e then
```
Comment (psafont) [inline src/alcotest-engine/test.ml:250]: Shouldn't this be `if not (exnp e) then`? It's confusing that this isn't tested and semantic expectations of the boolean returned by the function aren't documented anywhere

**Pull request merged**